src/parser.yy

```diff
@@ -362,7 +362,9 @@ macros:
 
 macro:
                 MACRO IDENT "(" macro_args ")" block_expr { $$ = driver.ctx.make_node<ast::Macro>($2, std::move($4), $6, @$); }
-        |       MACRO IDENT "(" ")" block_expr { $$ = driver.ctx.make_node<ast::Macro>($2, ast::ExpressionList{}, $5, @$); }
+        |       MACRO IDENT "(" ")" block_expr            { $$ = driver.ctx.make_node<ast::Macro>($2, ast::ExpressionList{}, $5, @$); }
```

Comment:
I believe the function args have a node that produces an optionally empty list. Did you want to have a `macro_args` node? It would make this easier to understand IMO.